tests/v2/validation/charts/cisbenchmark_test.go

```diff
@@ -0,0 +1,298 @@
+		if err != nil {
+			return false, err
+		}
+
+		scanIdx := -1
+
+		for idx, scan := range scanList.Data {
+			if scan.Name == scanName {
+				scanIdx = idx
+				break
+			}
+		}
+
+		if scanIdx < 0 {
+			return false, errors.New("scan not found")
+		}
+
+		clusterScanStatusType := &charts.ClusterScanStatus{}
+		scanStatus := scanList.Data[scanIdx].Status
+		err = v1.ConvertToK8sType(scanStatus, clusterScanStatusType)
+		if err != nil {
+			return false, err
+		}
+
+		if clusterScanStatusType.Display.State == scanError {
+
+			return false, errors.New(clusterScanStatusType.Display.Message)
+		} else if clusterScanStatusType.Display.State == scanPending ||
+			clusterScanStatusType.Display.State == scanRunning ||
+			clusterScanStatusType.Display.State == scanReporting {
+
+			logrus.Infof("cluster scan is %s", clusterScanStatusType.Display.State)
+			return false, nil
+		}
+
+		return true, nil
+	})
+	require.NoError(c.T(), err)
+
+	c.CheckScanReport(steveClient, scanName)
+}
+
+func (c *CisBenchmarkTestSuite) CheckScanReport(steveClient *v1.Client, scanName string) {
+	scanReportList, err := steveClient.SteveType(charts.ClusterScanReportType).List(nil)
+	require.NoError(c.T(), err)
+
+	scanReportIdx := -1
+
+	for idx, scanReport := range scanReportList.Data {
+		if strings.Contains(scanReport.Name, scanName) {
+			scanReportIdx = idx
+			break
+		}
+	}
+
+	require.GreaterOrEqualf(c.T(), scanReportIdx, 0, "scan report not found")
+	cisReportSpec := &charts.ClusterScanReportSpec{}
+	scanReportSpec := scanReportList.Data[scanReportIdx].Spec
+	err = v1.ConvertToK8sType(scanReportSpec, cisReportSpec)
+	require.NoError(c.T(), err)
+
+	reportData := &charts.CisReport{}
+	err = json.Unmarshal([]byte(cisReportSpec.ReportJson), &reportData)
+	require.NoError(c.T(), err)
+	logrus.Infof("Out of total number of %d scans, %d scans passed, %d scans were skipped and %d scans failed", reportData.Total, reportData.Pass, reportData.Skip, reportData.Fail)
+	for _, group := range reportData.Results {
+		for _, check := range group.Checks {
+			assert.NotEqualf(c.T(), "fail", check.State, "check failed: id: %s state: %s description: %s ", check.Id, check.State, check.Description)
+		}
+	}
+
+	require.Equalf(c.T(), reportData.Fail, 0, "cluster scan failed")
+}
+
+func (c *CisBenchmarkTestSuite) TestUpgradeCisbenchmarkChart() {
+	subSession := c.session.NewSession()
+	defer subSession.Cleanup()
+
+	client, err := c.client.WithSession(subSession)
+	require.NoError(c.T(), err)
+
+	// Change cisbenchmark install option version to previous version of the latest version
+	versionsList, err := client.Catalog.GetListChartVersions(charts.RancherCisBenchmarkName, catalog.RancherChartRepo)
+	require.NoError(c.T(), err)
+
+	if len(versionsList) < 2 {
+		c.T().Skip("Skipping the upgrade case, only one version of cis-benchmark is available")
+	}
+	assert.GreaterOrEqualf(c.T(), len(versionsList), 2, "There should be at least 2 versions of the cis-benchmark chart")
```

Comment:
I've not seen testing.Skip used before. Does it not stop the test if that condition is met? If so, what is the purpose of the assert?

Reply:
Correct. The CIS Chart upgrade test scenario will be skipped if only one version of the chart is available in the list of versions. For example on release-v2.8 only one version of CIS Benchmark chart is available as of now. Hence, the upgrade test scenario will be skipped.